Makefile

```diff
@@ -82,7 +82,7 @@ GOLANGCI_LINT = $(shell pwd)/bin/golangci-lint
 golangci-lint:
 	@[ -f $(GOLANGCI_LINT) ] || { \
 	set -e ;\
-	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell dirname $(GOLANGCI_LINT)) v1.29.0 ;\
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell dirname $(GOLANGCI_LINT)) v1.37.1 ;\
```

Comment:
we need to have the ci https://github.com/kubernetes-sigs/kubebuilder/blob/master/.github/workflows/ci.yml#L22 using the same version. Could you please check and update all to use the latest version supported by the GitHub action?

Reply:
When searched across the repo for "golangci-lint" or "v1.29", the only occurrences are Makefile, Contributing.md and  ci.yml. I guess it's up to date now